consumergroup/consumer_group.go

```diff
@@ -172,11 +177,13 @@ func JoinConsumerGroup(name string, topics []string, zookeeper []string, config
 	}
 
 	// Register itself with zookeeper
-	if err := cg.instance.Register(topics); err != nil {
-		cg.Logf("FAILED to register consumer instance: %s!\n", err)
-		return nil, err
-	} else {
-		cg.Logf("Consumer instance registered (%s).", cg.instance.ID)
+	if cg.existInstance() {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought:** ‏We use `cg.existInstance()` as a check before we use `cg.instance` without guarding with mutexes, so it seems to me that it is possible `cg.existInstance()` to return true, another goroutine to set `cg.instance` to nil and then when we invoke `cg.instance.Register(topics )` we panic.

Reply:
is a zero sum, and it'd not work because we would be locking the check but not where is being overwritten,(actually if we knew we would not need this check) till this moment we still don't know where this happens, this is a last resort solution, that works by luck.